src/components/detail/detail.component.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { MarginProps } from "styled-system";
+import { IconType } from "../icon";
+import { filterStyledSystemMarginProps } from "../../style/utils";
+import tagComponent from "../../__internal__/utils/helpers/tags/tags";
+import {
+  StyledDetail,
+  StyledDetailContent,
+  StyledDetailIcon,
+  StyledDetailFootnote,
+} from "./detail.style";
+
+export interface DetailProps extends MarginProps {
```

Comment:
Since we are using `tagComponent()` below, we'll want to extend the `TagProps` interface so we are accounting for the `data-element` and `data-role` props.

```suggestion
import { TagProps } from "../../__internal__/utils/helpers/tags";
export interface DetailProps extends MarginProps, TagProps {
```